core/email.py

```diff
@@ -246,29 +250,31 @@ def sendEMail(*,
     return True
 
 
-def sendEMailToAdmins(subject: str, body: str, *args, **kwargs):
+def sendEMailToAdmins(subject: str, body: str, *args, **kwargs) -> bool:
     """
-        Sends an e-mail to the root users of the current app.
+    Sends an e-mail to the root users of the current app.
+
+    If conf["viur.email.admin_recipients"] is set, these recipients
+    will be used instead of the root users.
 
-        :param subject: Defines the subject of the message.
-        :param body: Defines the message body.
+    :param subject: Defines the subject of the message.
+    :param body: Defines the message body.
     """
     success = False
     try:
-        if "user" in dir(conf["viur.mainApp"]):
-            users = []
-            for userSkel in conf["viur.mainApp"].user.viewSkel().all().filter("access =", "root").fetch():
-                users.append(userSkel["name"])
-
-            if users:
-                ret = sendEMail(dests=users, stringTemplate=os.linesep.join((subject, body)), *args, **kwargs)
-                success = True
-                return ret
-            else:
-                logging.warning("There are no root-users.")
-
-    except Exception:
-        raise
+        users = []
+        if conf["viur.email.admin_recipients"] is not None:
+            users = normalize_to_list(conf["viur.email.admin_recipients"])
+        elif "user" in dir(conf["viur.mainApp"]):
+            for user_skel in conf["viur.mainApp"].user.viewSkel().all().filter("access =", "root").fetch():
+                users.append(user_skel["name"])
+
+        if users:
+            ret = sendEMail(dests=users, stringTemplate=os.linesep.join((subject, body)), *args, **kwargs)
+            success = True
```

Comment:
Hmm what the heck is the purpose of this variable?
```suggestion
```
why isn't it just the Exception part?

Reply:
Because the case `success = False` can occur also without an exception (no users available). To unify this there is this variable. And as this has to be checked after both cases, this is in the `finally` block.